src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ActionPlanService.kt

```diff
@@ -71,10 +75,18 @@ class ActionPlanService(
     }
   }
 
-  private fun updateDraftActivityPlan(draftActionPlan: ActionPlan, update: ActionPlan) {
-    update.numberOfSessions?.let {
+  private fun updateDraftActivityPlan(
+    draftActionPlan: ActionPlan,
+    numberOfSessions: Int?,
+    activityUpdate: ActionPlanActivity?
+  ) {
+    numberOfSessions?.let {
       draftActionPlan.numberOfSessions = it
     }
+
+    activityUpdate?.let {
```

Comment:
is it possible to remove an activity using this api?

Reply:
i.e. is it clear in the contract that this is additive, as opposed to replacing all the activities on the action plan with content of the request body ?

Reply:
(this would affect the mutability of the list in the action plan entity, as changed above)

Reply:
Delete activity can't be specified using the current request object design. An update can be done but not implemented for skeleton. The request object reflects this.

data class UpdateActionPlanDTO(
  val numberOfSessions: Int?,
  val activity: CreateActionPlanActivityDTO?
)

However, I can make more clear by change the name of the variable derived from activity to newActivity rather than activityUpdate. This would give clarity. Like it.

Reply:
Done